src/cpp/server/server_context.cc

```diff
@@ -283,12 +302,15 @@ void ServerContext::TryCancel() const {
 }
 
 bool ServerContext::IsCancelled() const {
-  if (has_notify_when_done_tag_) {
-    // when using async API, but the result is only valid
+  if (completion_tag_) {
```

Comment:
well, doesn't look like the actual logic needed to be changed.. just comments

Reply:
Yes, that's right. I wanted to make the comments clear since the same code is called in both cases but the time when it is valid is different between the two.